bindings_ffi/src/mls.rs

```diff
@@ -163,10 +163,11 @@ impl FfiSignatureRequest {
         let mut inner = self.inner.lock().await;
         let signature_text = inner.signature_text();
         inner
-            .add_signature(Box::new(RecoverableEcdsaSignature::new(
-                signature_text,
-                signature_bytes,
-            )))
+            .add_signature(Box::new(
```

Comment:
This is the problem. We don't need to `sign_with_legacy_key` here. The code you wrote in the `IdentityStrategy` will sign with the legacy key automatically, and there will be no signature request at all in that case. This method is for cases where we _need_ a wallet signature.